packages/hoppscotch-common/src/helpers/RequestRunner.ts

```diff
@@ -492,23 +492,11 @@ export function runRESTRequest$(
 }
 
 function updateEnvsAfterTestScript(runResult: E.Right<SandboxTestResult>) {
-  const updatedGlobalEnvVariables = updateEnvironments(
-    // @ts-expect-error Typescript can't figure out this inference for some reason
-    cloneDeep(runResult.right.envs.global),
-    "global"
-  )
-
-  const updatedSelectedEnvVariables = updateEnvironments(
-    // @ts-expect-error Typescript can't figure out this inference for some reason
-    cloneDeep(runResult.right.envs.selected),
-    "selected"
-  )
-
   const updatedRunResult = {
     ...runResult.right,
     envs: {
-      global: updatedGlobalEnvVariables,
-      selected: updatedSelectedEnvVariables,
+      global: runResult.right.envs.global,
+      selected: runResult.right.envs.selected,
     },
   }
```

Comment:
`updatedRunResult` can be removed given it has evolved to be the same as `runResult`. Also, the `updateEnvsAfterTestScript()` function can be updated to remove the return value given the current consumers don't expect a value.